### ISSUE TITLE
Diagnostics summit 2019-03-07 and 2019-03-08

### ISSUE THREAD

**Issue by hashseed** [nodejs/diagnostics#203]
As discussed in another [thread](https://github.com/nodejs/diagnostics/issues/201), I would like to host a diagnostics summit at the [Google office in Munich](https://goo.gl/maps/A8qrggPWRfq) early 2019. I'm hoping that having this event in Europe could allow some of us who do not have the means to travel to North America to attend.

I booked conference rooms for 2019-03-07 and 2019-03-08 at the Google office in Munich.

[Info and agenda](https://docs.google.com/document/d/1Ub4kqznhEXHzKtS4ArK5KGBzwRLa29Kamr4lo2BvzcE/edit#heading=h.2uhgwd45mx6v).

List of attendees (so far):
- addaleax
- bmeurer
- Flarna
- gireeshpunathil 
- hashseed
- jasnell
- joyeecheung (tentatively)
- juliangruber
- mcollina
- mhdawson
- misterdjules
- mmarchini
- ofrobots
- Qard
- yunong
- watson
- BridgeAR

**Comment by mcollina** [comment]
Definitely +1. Considering how much advance we have, I do not see a problem in picking dates right now. I'd personally go with February to lower the gap between the summits.

**Comment by mhdawson** [comment]
Feb sounds good to me.

**Comment by yunong** [comment]
@hashseed any interest in hosting something in September instead?

**Comment by hashseed** [comment]
September is a bit too short-notice for me personally, and I was under the impression that most people involved would attend JS Interactive and the attached Node.js collaborator summit and meet each other face to face anyways.

I was rather going for mirroring the Ottawa meeting and it seems to me that an annual cadence is a good choice. There are no major conferences before and after February so there is no risk of overlap.

In the mean time I was able to book the facilities in Munich for February 20th and 21st. I'm still waiting for a final confirmation as to whether we can definitely plan around these dates yet. But please make sure that you block out these dates on your calendars.

**Comment by hashseed** [comment]
I got confirmation that February 20th and 21st are set dates.

**Comment by hashseed** [comment]
I would like to ask diagnostics WG members to give an indication whether they want to attend.

**Comment by mcollina** [comment]
I’ll be there

**Comment by mmarchini** [comment]
I'll be there too

**Comment by ofrobots** [comment]
I should be able to make it.

**Comment by hashseed** [comment]
I am moving the diagnostics summit to March 7th and 8th to not collide with [AgentConf](https://www.eventbrite.de/e/agentconf-2019-tickets-42359722069) and also to move it closer to [JSKongress](https://twitter.com/JSKongress), which happens on the Monday/Tuesday the week after.

I hope that's alright with everyone, including ones who already signed up (@mcollina, @ofrobots, @mmarchini, @bmeurer), and sorry that I had to change plans. It's still 8 months to go after all :)

**Comment by Flarna** [comment]
I have planed to attend.

**Comment by misterdjules** [comment]
@hashseed I would like to attend this diagnostics summit.

**Comment by gireeshpunathil** [comment]
I too will plan to attend.

**Comment by yunong** [comment]
I plan to attend

**Comment by mcollina** [comment]
Is this confirmed? Should we start booking plane tickets, hotels etc?

**Comment by Qard** [comment]
Is there a reason for having it the week before JSKongress rather than the days right after? It'd be nice for international folks to be able to just fly in for a week and catch a conference and diag summit in one go.

**Comment by hashseed** [comment]
Hold on for a few days. I will figure out whether moving it to after JSKongress is an option.

In the meantime, we can start collecting agenda items.

**Comment by hashseed** [comment]
Unfortunately we cannot move the event. The venue I booked for the 7th and 8th is already taken for the 13th and 14th. So I think these dates are as confirmed as we can get :)

**Comment by joyeecheung** [comment]
@hashseed Is it possible for the host to provide an invitation letter for the visa application? As I understand if the venue is not rented by the foundation, then the foundation will not be able to provide visa letters.

**Comment by hashseed** [comment]
Invitation letter should not be an issue. Let's take this offline.

**Comment by mmarchini** [comment]
I'm not entirely sure I'll make it due to other travel arrangements. When's the deadline to confirm attendance?

**Comment by jasnell** [comment]
Added myself to the attendee list. See you all there.

**Comment by misterdjules** [comment]
@hashseed FWIW, created a [Google doc](https://docs.google.com/document/d/1Ub4kqznhEXHzKtS4ArK5KGBzwRLa29Kamr4lo2BvzcE/edit?usp=sharing) to start collaborating on an agenda.

**Comment by mhdawson** [comment]
I'll be looking to firm up travel approval early in the new year.

**Comment by mike-kaufman** [comment]
I've opened #267 to nail down the agenda (thought it would be more clear in a separate issue).

**Comment by mhdawson** [comment]
@addaleax you will be a great addition to the discussion :)

**Comment by watson** [comment]
Regarding what we talked about on yesterdays meeting: If we invite a few outside speakers to talk about their diagnostics challenges running Node.js in big deployments, would we be able to cover some of their travel costs?

**Comment by mhdawson** [comment]
@watson that is a good question. The current travel fund does not cover that so we might need to go to the board.  @MylesBorins is that something you can ask about?

For context at the Diagnostics Summit we are exploring if we can get somebody with a large deployment to present on what diagnostics tools/techniques they use in their deployments.  The question is if there is a way to cover some of their travel costs.

**Comment by watson** [comment]
@mhdawson @mike-kaufman the team at Uber I had contact to in Europe unfortunately switched from Node to Java and Go about a year ago, so no luck there.

**Comment by mhdawson** [comment]
Ok I'll reach out to @dshaw to see if somebody from Paypal might be available.

**Comment by mmarchini** [comment]
> the team at Uber I had contact to in Europe unfortunately switched from Node to Java and Go about a year ago

Would they be willing to chat with us about which tools they use for Diagnostics in Java and Go (either on our meeting calls or at the summit)?

**Comment by Flarna** [comment]
I asked around if someone here as a good contact to one of our customers with a big NodeJs deployment but I failed to get one.

> switched from Node to Java and Go about a year ago

Looks like Go is the new Node...

**Comment by mhdawson** [comment]
@hashseed just about to book travel.  Just want to confirm dates are fixed and we are good to go.

**Comment by hashseed** [comment]
Dates are very fixed. I already nailed down catering details too.

**Comment by Qard** [comment]
I’ll be there too. Apparently I forgot to mention that earlier. Sorry about that. 😅

**Comment by hashseed** [comment]
@Flarna do you or someone from Dynatrace want to give a short presentation on how you use Node.js and where the issues are?

**Comment by Flarna** [comment]
We don't use Node.js in the sense that we create applications using node.js as runtime. Our servers are implemented in Java.
We create agents which are injected in applications of our customers (Node.Js, Java, .NET,...) to extract data and send it to our servers and correlate it.
So there is no Node.JS deployment or similar I could describe. For privacy reasons I can't share any details about applications from our customers.

But I think I could give a rough overview what data our agent extracts and how. And maybe a comparison to our java agent and why they are able to provide more insight. I will discuss this internally before giving a final go here.

**Comment by hashseed** [comment]
Thanks. I think these insights are very valuable.

**Comment by mhdawson** [comment]
I agree knowing the gap (and possibly the reason why) between Java and Node.js would be interesting.

**Comment by hashseed** [comment]
If you are intending to attend, please add your name to the [agenda doc](https://docs.google.com/document/d/1Ub4kqznhEXHzKtS4ArK5KGBzwRLa29Kamr4lo2BvzcE/edit#heading=h.2uhgwd45mx6v) so that I can prepare name badges upfront. Thanks!

**Comment by devsnek** [comment]
will there be a way to remote in?

**Comment by hashseed** [comment]
We can definitely set up a live stream one way or another.

**Comment by devsnek** [comment]
@hashseed what about remote participation

**Comment by hashseed** [comment]
Would a public hangout solve that?

**Comment by grawk** [comment]
Hi all. I manage the Node.js Infrastructure team at PayPal. I'm trying to get a business justification case made to my management to send a team member to the summit.

**Comment by mike-kaufman** [comment]
FYI I've had some hard conflicts come up and I will not be able to attend.

**Comment by mhdawson** [comment]
@grawk sorry for the late reply, just catching up on issues today.  We've had a public hangout for the last summit but I don't think it works well for participation (@cjihrig could comment on that more specifically). Better than not participating at all, but hard to follow and actively participate.   So I'd suggest 2 things. 

* We can try to have a hangout for the sessions but want to set expectations as YMMV.  
* We were hoping for a presentation for PayPal on what you do in terms of Diagnostics and Problem 
  determination and having you do that via a Hangout would be good. Is that possible?

**Comment by cjihrig** [comment]
> @cjihrig could comment on that more specifically

I thought the hangout option worked well. If you're **very** talkative, then it's definitely not ideal, but better than nothing. The biggest issue is if the group splits up into multiple rooms and there is only a single hangout to attend.

**Comment by mhdawson** [comment]
@cjihrig good to hear that it worked better than I might have thought.

**Comment by Qard** [comment]
This is a Google office though. Should be set up well to set up a Hangout per-room easy enough, right?

**Comment by mhdawson** [comment]
@hashseed are the rooms setup for Hangouts?

**Comment by hashseed** [comment]
Yes. They do have GVC units. Hangout should be available.

**Comment by hashseed** [comment]
I'm going to reserve dinner for 20 people at a steakhouse closeby. Please let me know if you don't want to join. Also, unfortunately, my budget does not cover buying dinner for everyone.

**Comment by hashseed** [comment]
@mcollina @mmarchini @watson @kjin @psmarshall @bmeurer @flarna @grawk 

I semi randomly scheduled the presentations over the two time slots. Please let me know if that order works for you.

**Comment by Flarna** [comment]
The presentation regarding opentracing/opencensus/trace context and mine point to similar topics so maybe we should move mine to the first day next to/after opentracing/opencensus/trace context.
But this is not required.

**Comment by hashseed** [comment]
Swapped some talks around to move @Flarna's talk to the first day.

**Comment by hashseed** [comment]
Unfortunately the steakhouse rejected my reservation. I booked a [Greek restaurant](https://goo.gl/maps/WAb1e9gLBwF2) for Thursday, 6:30pm.

**Comment by grawk** [comment]
Hi @mhdawson I think 30 minutes would be reasonable. This would allow us to provide good content in a short timeframe.

**Comment by mhdawson** [comment]
@hashseed can you let @grawk know what slots ate avail so that he can plan join to present.

**Comment by hashseed** [comment]
For people who want to join remote...

Join via VC: meet.google.com/uwh-hiuf-mnn
Watch stream: stream.meet.google.com/stream/72f5f7ea-61e3-49a8-b6a4-36006d57e598

@grawk, could you join via VC this way to give your talk at 09:30 CET? You are of course welcome to join earlier.

**Comment by mhdawson** [comment]
@grawk understand 9:30 our time is early. Let us know when you have joined and we'll figure out how to fit you in.

**Comment by delvedor** [comment]
Hello!

I tried to join the streaming at `stream.meet.google.com/stream/72f5f7ea-61e3-49a8-b6a4-36006d57e598`, but once I open the page I get `This stream is not available to your domain`.

Any suggestion? Thanks!

**Comment by hashseed** [comment]
Yes, unfortunately stream does not work. Joining as VC should work though.

**Comment by hashseed** [comment]
VC link for today: https://meet.google.com/zfa-gzpe-hwe

**Comment by bmeurer** [comment]
Covered `async_hooks` discussion items regarding performance from yesterday in http://bit.ly/making-async-hooks-fast-enough, feel free to chime in if you wanna help (or have some thoughts). Just ask for edit/comment permissions.

**Comment by bmeurer** [comment]
And also for the discussion around surfacing engine signals: http://bit.ly/v8-tracing-signals

**Comment by mmarchini** [comment]
Action items: https://docs.google.com/document/d/1ftJCXFM_Yi45O2nr5O7YwdOMO7iUbWSx-UStvOYMu28/edit?usp=sharing

**Comment by hashseed** [comment]
Recordings can be found [here](https://drive.google.com/drive/folders/1YLrO8XwAdWPaFNmyZZRqPAsSIDGo_RXw). The audio seems pretty clear, but for some reason the video quality deteriorated with time, on the first day, so that the slides are somewhat unreadable after two hours.

**Comment by mhdawson** [comment]
@hashseed can we get the slides for all of the talks/session context presentations that were given so that we can add them to the repo somewhere?

This was mine: 
[Diagnostic Summit - Best Practices.pdf](https://github.com/nodejs/diagnostics/files/2957405/Diagnostic.Summit.-.Best.Practices.pdf)

**Comment by watson** [comment]
Below are deep links to the different recorded sessions and talks. Feel free to edit if I made a mistake.

### Talks

- [Intro to OpenCensus, OpenTracing, and TraceContext](https://drive.google.com/file/d/1Of-bwwZtv0sb-7R1cEjyIqizDgyT4a4i/view?t=865)
- [Perfetto and Node.js](https://drive.google.com/file/d/1Of-bwwZtv0sb-7R1cEjyIqizDgyT4a4i/view?t=2557)
- [Intro into APM agents, comparison of Node.js and Java](https://drive.google.com/file/d/1Of-bwwZtv0sb-7R1cEjyIqizDgyT4a4i/view?t=4700)
- [Zero cost async stack traces](https://drive.google.com/file/d/1Of-bwwZtv0sb-7R1cEjyIqizDgyT4a4i/view?t=6797)
- [Status of Async Hooks](https://drive.google.com/file/d/1Of-bwwZtv0sb-7R1cEjyIqizDgyT4a4i/view?t=9870)
- [How to use a core dump](https://drive.google.com/file/d/1Of-bwwZtv0sb-7R1cEjyIqizDgyT4a4i/view?t=11302)
- [Diagnostics Best Practices documentation](https://drive.google.com/file/d/1Of-bwwZtv0sb-7R1cEjyIqizDgyT4a4i/view?t=12125)
- [Diagnostics in Java](https://drive.google.com/file/d/15MzZoIiUKpOUr86inpUqsl2phUsk05x2/view?t=143)
- [Status of Node Clinic / Doctor, Bubbleprof, Flame](https://drive.google.com/file/d/15MzZoIiUKpOUr86inpUqsl2phUsk05x2/view?t=2008)
- [State of CPU Profiling in V8](https://drive.google.com/file/d/1vIdYiFYW7S3804U7_5iznP9M-IO8xvNv/view?t=52)
- [CloudFlare Workers](https://drive.google.com/file/d/1zlY37F-AtunGyqMbMC6TaAB40icN8GU0/view)

### Discussions

- [V8 Signals](https://drive.google.com/file/d/1LeSfIzArwvJfy2kGtbxF6HiMOukLabOg/view?t=281)
- Documentation best practices: _I couldn't find this video_
- [Diagnostics Tooling Support Tiers](https://drive.google.com/file/d/1LeSfIzArwvJfy2kGtbxF6HiMOukLabOg/view?t=3388)
- [Async Hooks](https://drive.google.com/file/d/1LeSfIzArwvJfy2kGtbxF6HiMOukLabOg/view?t=9227)
- [Post-mortem debugging](https://drive.google.com/file/d/1Enw2Z0R9pNsoaCVq8LPK6Ja-P-T5QX-h/view?t=2940)
- [Trace events, system tracing, profiling](https://drive.google.com/file/d/1U9SfKfb83M-AQX3IL-XgX-OGuGAz4c5w/view)
- [APM instrumentation](https://drive.google.com/file/d/11nk_cZYUCZhUKptHD-4dlS5PxQTeLXlc/view)

**Comment by hashseed** [comment]
Thanks for the video links! I only uploaded the videos for the morning of the second day a few hours ago. Please check again.

**Comment by joyeecheung** [comment]
Can we upload the sessions to the node.js youtube channel? (Not sure who has access to the account)

**Comment by hashseed** [comment]
No objections from my side.

**Comment by watson** [comment]
@hashseed 

> I only uploaded the videos for the morning of the second day a few hours ago. Please check again.

Thanks for making them available 😃 I think I managed to include them. At least I don't see any new videos since last I checked.
